app/api/server/v1/teams.ts

```diff
@@ -190,16 +192,35 @@ API.v1.addRoute('teams.listRoomsOfUser', { authRequired: true }, {
 API.v1.addRoute('teams.members', { authRequired: true }, {
 	get() {
 		const { offset, count } = this.getPaginationItems();
-		const { teamId, teamName } = this.queryParams;
-		const { query } = this.parseJsonQuery();
+		const { teamId, teamName, status, username, name } = this.queryParams;
+		const SecondaryOptionalParam = Match.Where((value) => {
+			if (!teamId) {
+				check(value, String);
+			}
+			return true;
+		});
+
+		check(this.queryParams, Match.ObjectIncluding({
+			teamId: Match.Maybe(String),
+			teamName: SecondaryOptionalParam,
+			status: Match.Maybe([String]),
+			username: Match.Maybe(String),
+			name: Match.Maybe(String),
+		}));
 
 		const team = teamId ? Promise.await(Team.getOneById(teamId)) : Promise.await(Team.getOneByName(teamName));
 		if (!team) {
 			return API.v1.failure('team-does-not-exist');
 		}
 		const canSeeAllMembers = hasPermission(this.userId, 'view-all-teams', team.roomId);
 
-		const { records, total } = Promise.await(Team.members(this.userId, team._id, canSeeAllMembers, { offset, count }, { query }));
+		const query = {
+			username: username ? new RegExp(username, 'i') : undefined,
+			name: name ? new RegExp(name, 'i') : undefined,
+			status: status ? { $in: status } : undefined,
+		} as Partial<FindOneOptions<IUser>>;
```

Comment:
You forgot to change this coercion 🙈

Reply:
Ohhh, you're right 😬. That's now fixed 
The `SecondOptionalParam` check was also giving some errors in the tests -- so I decided to remove it until we can think of a better way to check this kind of params (maybe in V2).